geode-core/src/main/java/org/apache/geode/internal/statistics/HostStatHelper.java

```diff
@@ -32,28 +32,32 @@
 public class HostStatHelper {
   private static final int PROCESS_STAT_FLAG = 1;
   private static final int SYSTEM_STAT_FLAG = 2;
+  private static OSVerifier osVerifier;
 
   static {
-    new OSVerifier();
+    osVerifier = new OSVerifier();
```

Comment:
In general, non-final statics are bad. Also, this line is simple enough that I would recommend you inline this static initializer and make osVerifier final:
```
private static final OSVerifier osVerifier = new OSVerifier();
```
Another option to consider (I'm not actually requesting this but I would help steer/review it if you choose to do this) is to change HostStatHelper from a utils class of static methods to an instance based object with non-static fields. This would make for better object design and improve testability of both the stat sampler as well as HostStatHelper.

For example, if HostStatHelper was renamed to OsStatisticsProvider, then GemFireStatSampler could change from:
```
Statistics stats =
    HostStatHelper.newProcess(getOsStatisticsFactory(), id, statName + "-proc");
processStats = HostStatHelper.newProcessStats(stats);
```
...to:
```
private final OsStatisticsProvider osStatisticsProvider = new OsStatisticsProvider();

Statistics stats = osStatisticsProvider.newProcess(getOsStatisticsFactory(), id, statName + "-proc");
processStats = osStatisticsProvider.newProcessStats(stats);
```
And then have the `OsStatisticsProvider` constructor read any System Properties to determine OS type.

Another good option is to have HostStatHelper and OsStatisticsFactory merge into a single class with no statics.

One major benefit to changing from statics: you can easily set a System Property in setUp for a test, then create the stat sampler or HostStatHelper/OsStatisticsFactory or any of these classes you intend to unit test. Since there's nothing that's `static final`, you can easily change the OS for many different tests in the same test class. When GemFireStatSampler executes the line `osStatisticsProvider = new OsStatisticsProvider();` the System Property would be read in the constructor of `OsStatisticsProvider`.

Reply:
> change HostStatHelper from a utils class of static methods to an instance based object with non-static fields

I like this idea, I also thought about it when I was dealing with the testing. And then the OsVerifier class will not be needed.